### PR TITLE
#1938 Added Create rig functionality

### LINKED ISSUE #1938: Create rig
Implement a create rig functionality, just like "New Log" when right clicking on Logs in the sidebar.
Reuse RigPropertiesModal by using PropertiesModalMode.

Split from WE-336

### PR DESCRIPTION
**Pull request by jaroslavbliznak**
[//]: # (Request Template Witsml Explorer)

[//]: # (Thank you for contributing to WITSML Explorer! Before submitting this PR, please fill in the following template.)

## Fixes

[//]: # (Write the GitHub issue number starting with #, or, for Equinor only, the Jira issue number starting with WE-)

This pull request fixes #1938 

## Description

[//]: # (Please include a written summary of the changes and which issue is fixed. List any dependencies that are required for this change._)

## Type of change

[//]: # (Mark any of the types of change that apply.)

* [ ] Bugfix
* [x] New feature (non-breaking change which adds functionality)
* [ ] Enhancement of existing functionality
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] This change requires a documentation update

## Impacted Areas in Application

[//]: # (List general components of the application that this PR will affect)

* [x] Frontend
* [x] API
* [ ] WITSML
* [ ] Other (please describe)

## Checklist:

[//]: # (Please tick all the boxes or remove the ones that aren't needed and explain why)

*Communication*
* [ ] I have made corresponding changes to the documentation
* [ ] PR affects application security

*Code quality*
* [ ] I have self-reviewed my code
* [x] No new warnings are generated

*Test coverage*
* [x] New code is covered by passing tests

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: Src/WitsmlExplorer.Frontend/components/Modals/RigPropertiesModal.tsx
Comment (janburak) [inline Src/WitsmlExplorer.Frontend/components/Modals/RigPropertiesModal.tsx:1]: Errors and helper texts in TextFields for nullable properties (all except name) should not be shown in create mode

File: Src/WitsmlExplorer.Api/Query/RigQueries.cs
Comment (janburak) [inline Src/WitsmlExplorer.Api/Query/RigQueries.cs:1]: .NullIfEmpty() should be used for all nullable strings (all except name)

File: Src/WitsmlExplorer.Frontend/components/ContextMenus/RigsContextMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      typeRig: "unknown",
+      yearEntService: null
+    };
+    const rigPropertiesModalProps: RigPropertiesModalProps = { mode: PropertiesModalMode.New, rig: newRig, dispatchOperation };
+    const action: DisplayModalAction = { type: OperationType.DisplayModal, payload: <RigPropertiesModal {...rigPropertiesModalProps} /> };
+    dispatchOperation(action);
+  };
+
+  return (
+    <ContextMenu
```
Comment (janburak) [inline Src/WitsmlExplorer.Frontend/components/ContextMenus/RigsContextMenu.tsx:64]: The context menu is missing refresh and paste functions (see LogsContextMenu).

**Comment by janburak** [comment]
Please fill out the issue number under "Fixes" in the PR template to link this PR to its respective issue

File: Src/WitsmlExplorer.Frontend/components/ContextMenus/RigsContextMenu.tsx (hunk 1 above)
Comment (jaroslavbliznak) [reply Src/WitsmlExplorer.Frontend/components/ContextMenus/RigsContextMenu.tsx:64]: Added paste and refresh functions

File: Src/WitsmlExplorer.Api/Query/RigQueries.cs
Comment (jaroslavbliznak) [reply Src/WitsmlExplorer.Api/Query/RigQueries.cs:1]: Fixed

File: Src/WitsmlExplorer.Frontend/components/Modals/RigPropertiesModal.tsx
Comment (jaroslavbliznak) [reply Src/WitsmlExplorer.Frontend/components/Modals/RigPropertiesModal.tsx:1]: Fixed on create mode is validate only name and uid

**Review by janburak — DISMISSED** [review]
- [X] Read issue and PR
- [X] Pulled branch and manually tested
- [X] Verify that PR resolves issue
- [X] Reviewed the code

**Review by janburak — DISMISSED** [review]

**Comment by janburak** [comment]
There seems to be an issue with approving PRs https://github.com/orgs/community/discussions/58535
Give this comment a thumbs up and I will merge the PR.

**Pull request merged**